File: exemplo.go
package main

import (
	"crypto/hmac"
	"crypto/sha512"
	"encoding/hex"
	"fmt"
	"io/ioutil"
	"net/http"
	"net/url"
	"strconv"
	"strings"
	"time"
)

func main() {

	// Constantes
	mbTapiID := "<chave_tapi>"
	mbTapiSecret := "<segredo>"
	requestHost := "https://www.mercadobitcoin.net"
	requestPath := "/tapi/v3/"

	// Nonce
	tapiNonce := strconv.FormatInt(time.Now().Unix(), 10)
	fmt.Println("Nonce:", tapiNonce)

	// Parâmetros
	params := url.Values{}
	params.Add("tapi_method", "get_account_info")
	params.Add("tapi_nonce", tapiNonce)

	// Gerar MAC
	paramsString := requestPath + "?" + params.Encode()
	h := hmac.New(sha512.New, []byte(mbTapiSecret))
	h.Write([]byte(paramsString))
	tapiMac := hex.EncodeToString(h.Sum(nil))

	// Realizar requisição POST
	request, err := http.NewRequest("POST", requestHost+requestPath, strings.NewReader(params.Encode()))

	// Gerar cabeçalho da requisição
	request.Header.Set("TAPI-ID", mbTapiID)
	request.Header.Set("TAPI-MAC", tapiMac)
	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")

	client := &http.Client{}
	resp, err := client.Do(request)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	// Exibindo dados da resposta no console
	fmt.Println("response Status:", resp.Status)
	body, _ := ioutil.ReadAll(resp.Body)
	fmt.Println("response Body:", string(body))
}

File: exemplo.js
const https = require('https');
const crypto = require('crypto');
const querystring = require('querystring');

// Constantes
const MB_TAPI_ID = '<chave_tapi>';
const MB_TAPI_SECRET = '<segredo>';
const REQUEST_HOST = 'www.mercadobitcoin.net';
const REQUEST_PATH = '/tapi/v3/';

// Nonce
let tapi_nonce = Math.floor(Date.now() / 1000);

// Parâmetros
let params = {
    'tapi_method': 'get_account_info',
    'tapi_nonce': tapi_nonce
};
params = querystring.stringify(params);

// Gerar MAC
let params_string = REQUEST_PATH + '?' + params;
let hash = crypto.createHmac('sha512', MB_TAPI_SECRET);
hash.update(params_string);
let tapi_mac = hash.digest('hex');

// Gerar cabeçalho da requisição
let headers = {
    'Content-Type': 'application/x-www-form-urlencoded',
    'TAPI-ID': MB_TAPI_ID,
    'TAPI-MAC': tapi_mac
};

// Parâmetros da requisição POST
const options = {
    hostname: REQUEST_HOST,
    path: REQUEST_PATH,
    method: 'POST',
    headers: headers
};
// Realizar requisição POST
const req = https.request(options, (res) => {
    let data = '';
    res.on('data', (chunk) => {
        data += chunk;
    });

    res.on('end', () => {
        // Exibindo resposta no console
        console.log(JSON.parse(data));
    })

}).on('error', (err) => {
    console.log("Error: " + err.message)
});
// Passando parâmetros
req.write(params)
req.end();